src/test/java/jenkins/plugins/logstash/persistence/ElasticSearchDaoTest.java

```diff
@@ -53,105 +56,67 @@ public void after() throws Exception {
     verifyNoMoreInteractions(mockHttpClient);
   }
 
-  @Test(expected = IllegalArgumentException.class)
-  public void constructorFailNullHost() throws Exception {
-    try {
-      createDao(null, 8200, "logstash", "username", "password");
-    } catch (IllegalArgumentException e) {
-      assertEquals("Wrong error message was thrown", "host name is required", e.getMessage());
-      throw e;
-    }
-  }
-
-  @Test(expected = IllegalArgumentException.class)
-  public void constructorFailEmptyHost() throws Exception {
-    try {
-      createDao(" ", 8200, "logstash", "username", "password");
-    } catch (IllegalArgumentException e) {
-      assertEquals("Wrong error message was thrown", "host name is required", e.getMessage());
-      throw e;
-    }
-  }
-
-  @Test(expected = IllegalArgumentException.class)
-  public void constructorFailMissingScheme() throws Exception {
-    try {
-      createDao("localhost", 8200, "logstash", "username", "password");
-    } catch (IllegalArgumentException e) {
-      assertEquals("Wrong error message was thrown", "host field must specify scheme, such as 'http://'", e.getMessage());
-      throw e;
-    }
-  }
-
-  @Test(expected = IllegalArgumentException.class)
-  public void constructorFailNullKey() throws Exception {
-    try {
-      createDao("http://localhost", 8200, null, "username", "password");
-    } catch (IllegalArgumentException e) {
-      assertEquals("Wrong error message was thrown", "elastic index name is required", e.getMessage());
-      throw e;
-    }
-  }
-
-  @Test(expected = IllegalArgumentException.class)
-  public void constructorFailEmptyKey() throws Exception {
+  @Test
+  public void constructorFailInvalidUrl() throws Exception {
     try {
-      createDao("http://localhost", 8200, " ", "username", "password");
+      createDao("localhost:8000/logstash", "username", "password");
+      fail("Expected an IllegalArgumentException.");
     } catch (IllegalArgumentException e) {
-      assertEquals("Wrong error message was thrown", "elastic index name is required", e.getMessage());
-      throw e;
+      assertEquals("Wrong error message was thrown", "java.net.MalformedURLException: unknown protocol: localhost", e.getMessage());
```

Comment:
minor: http://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.html looks like a better idea here

Reply:
I consider using ExpectedException as a bad practice (see also https://artemzin.com/blog/test-expected-badpracticeexception/)
Why rethrow when we already catched it and checked the message is what we expect.

Reply:
Yeah, but I was talking about the ExpectedException rule
```

     @Rule
     public ExpectedException thrown= ExpectedException.none();
// ...
     thrown.expect(NullPointerException.class);
     thrown.expectMessage("happened");
     throw new NullPointerException("What happened?");
```